src/main/java/com/gradle/Utils.java

```diff
@@ -82,29 +82,12 @@ static String stripPrefix(String prefix, String string) {
         return string.startsWith(prefix) ? string.substring(prefix.length()) : string;
     }
 
-    static String appendIfMissing(String str, String suffix) {
-        return str.endsWith(suffix) ? str : str + suffix;
-    }
-
-    static URI appendPathAndTrailingSlash(URI baseUri, String path) {
-        if (isNotEmpty(path)) {
-            String normalizedBasePath = appendIfMissing(baseUri.getPath(), "/");
-            String normalizedPath = appendIfMissing(stripPrefix("/", path), "/");
-            return baseUri.resolve(normalizedBasePath).resolve(normalizedPath);
-        }
-        return baseUri;
+    static String prependIfMissing(String suffix, String str) {
+        return str.startsWith(suffix) ? str : suffix + str;
     }
 
-    static String concatenatePaths(String basePath, String path) {
-        if (isNotEmpty(basePath)) {
-            if (isNotEmpty(path)) {
-                String normalizedBasePath = appendIfMissing(basePath, "/");
-                String normalizedPath = stripPrefix("/", path);
-                return normalizedBasePath + normalizedPath;
-            }
-            return basePath;
-        }
-        return path;
+    static String appendIfMissing(String str, String suffix) {
```

Comment:
I find it surprising that `appendIfMissing` and `prependIfMissing` swap the order of parameters. I think the parameters are trying to match the order of the resulting string, but for two methods that are doing very similar things, I would expect the method signatures to match.

Reply:
I thought about this, but I find the calling code easier to read the current way.

Reply:
That's fine. Just an opinion in this case 🙂

Reply:
I've applied your suggestion. Looks indeed better.